dali/operators/reader/loader/video/video_file.h

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_VIDEO_VIDEO_FILE_H_
+#define DALI_OPERATORS_READER_LOADER_VIDEO_VIDEO_FILE_H_
+
+extern "C" {
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libavutil/avutil.h>
+#include <libswscale/swscale.h>
+}
+
+#include "dali/core/common.h"
+
+#include <vector>
+#include <string>
+
+
+namespace dali {
+struct IndexEntry {
+  int last_keyframe_id;
+  int pts;
+  bool is_keyframe;
+  bool is_flush_frame;
+};
+
+class DLL_PUBLIC VideoFileCPU {
+ public:
+  VideoFileCPU(std::string &filename);
+
```

Comment:
Please delete copy constructor and assignment and delete (or implement) move.

Reply:
There is a `unique_ptr` field now that would impact this, but I used `DISABLE_COPY_MOVE_ASSIGN` for clarity.